docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,399 @@
+
+## Production of protected container image
+
+In the confidential containers V1 design, after the user completes the production of the conventional container image, it is necessary to add security protection functions for the container image, including encryption and signing. It should be noted that the encryption process must be carried out before the signing process.
+
+### Image encryption
+
+Image encryption is based on layer granularity. Users can selectively encrypt layers containing private data or choose not to encrypt any layers. The high-level process of encrypting an image layer is as follows:
+
+1. Dynamically generate a random symmetric key.
+
+2. Encrypt container image layer with symmetric key.
+
+3. Encrypt the symmetric key with the user's private key.
+
+4. Write the symmetric key encrypted by the user's private key into the container image manifest as the content of the annotation of the container image layer.
+
+The above process needs to be carried out according to the container image encryption format specification defined in [containers/ocicrypt](https://github.com/containers/ocicrypt). In the confidential container V1 architecture, image-rs calls the ocicrypt-rs's APIs for specific implementation.
+
+Details are briefly described below:
+
+After dynamically generating the random symmetric key and encrypting the image layer, two special JSON structures need to be generated, One is used to record public encrypted information (such as encryption algorithm), which is called PublicLayerBlockCipherOptions, and the other is used to record secret information (such as symmetric key used for encryption), which is called PrivateLayerBlockCipherOptions. The formats are as follows:
+
+- PublicLayerBlockCipherOptions:
+
+  ```json
+  {
+      "cipher": "AES_256_CTR_HMAC_SHA256",
+      "hmac": “M0M5OTA5QUZFQzI1MzU0RDU1MURBR…”
+      "cipheroptions": {}
+  }
+  ```
+
+- PrivateLayerBlockCipherOptions:
+
+  ```json
+  {
+      "symkey": "54kiln1USEaKnlYhKdz+aA==",
+      "cipheroptions": {
+          "nonce": "AdcRPTAEhXx6uwuYcOquNA=="
+      }
+  }
+  ```
+
+Then, encrypt the PrivateLayerBlockCipherOptions with the user's private key. Under the design of confidential containers V1, when decrypting the image, the decryption of PrivateLayerBlockCipherOptions needs to be carried out through the key provider protocol with the attestation-agent. Therefore, when encrypting the image, thePrivateLayerBlockCipherOptions also needs to be encrypted through the key provider protocol, Moreover, the name of the key provider program used in encryption needs to be consistent with that used in decryption. Under the design of confidential containers V1, this name is uniformly specified as "attestation-agent".
```

Comment:
missing space in thePrivateLayerBlockCipherOptions.

Also consider backticks around PrivateLayerBlockCipherOptions (here and other places).